src/www/ui/api/Controllers/UploadController.php

```diff
@@ -837,4 +837,54 @@ public function getGroupsWithPermissions($request, $response, $args)
     $res["permGroups"] = $finalPermGroups;
     return $response->withJson($res, 200);
   }
+
+  /**
+   * Set the main license for the upload
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function setMainLicense($request, $response, $args)
+  {
+    try {
+      global $PG_CONN;
+      $uploadId = intval($args['id']);
+      $body = $this->getParsedBody($request);
+      $shortName = $body['shortName'];
+      $returnVal = null;
+      $license = null;
+      $licenseDao = $this->container->get('dao.license');
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadId)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if (empty($shortName)) {
+        $returnVal = new Info(400, "Short name missing from request.",
+          InfoType::ERROR);
+      } else {
+        $license = $licenseDao->getLicenseByShortName($shortName,
+          $this->restHelper->getGroupId());
+
+        if ($license === null) {
+          $returnVal = new Info(404, "No license with shortname '$shortName' found.",
+            InfoType::ERROR);
+        } else if (0 < (intval($this->dbHelper->getDbManager()->getSingleRow("SELECT COUNT(*) FROM upload_clearing_license WHERE rf_fk= $1 AND upload_fk = $2 AND group_fk = $3", [$license->getId(), $uploadId, $this->restHelper->getGroupId()],
```

Comment:
Better use the existing function `ClearingDao::getMainLicenseIds()` and then use `in_array()` to check if license exists, rather than writing a SQL query.